sources/node2/KalturaClientBase.js

```diff
@@ -334,16 +335,20 @@ class RequestBuilder extends kaltura.VolatileRequestData {
 					sessionId = response.headers[header];
 				}
 			}
-			client.debug('Response server [' + serverId + '] session [' + sessionId + ']: ' + data);
-
+			
 			let json;
-			try {
-				json = JSON.parse(data);
-			} catch (err) {
-				json = {
-					error: err
+			if (typeof data === 'string') {
+				client.debug('Response server (Str) [' + serverId + '] session [' + sessionId + ']: ' + data);
+				try {
+					json = JSON.parse(data);
+				} catch (err) {
+					json = data; //handle cases where Kaltura returns non JSON formatted responses like KS or serve
 				}
+			} else {
+				client.debug('Response server (Json) [' + serverId + '] session [' + sessionId + ']: ' + JSON.stringify(data));

```

Comment:
Hi @zoharbabin - This is exactly the same as line 341 only you call `JSON.stringify(data)` rather than output `data` as is.
Best do only that in the `if` block and call `client.debug()` after the condition is evaluated.

Reply:
The if comment is also different (Str) vs. (Json).
I don't mind it either way, just think this is cleaner, as these lines are a bit confusing as it is...